Issue #235: add some examples to root cli commands entry (#427)
redhat-documentation/supplementary-style-guide: supplementary_style_guide/style_guidelines/code-commands.adoc

Some commands require root privileges to run. Users without root privileges must first do one of the following to run such a command:

* Preface the command with `sudo` to temporarily change their current privileges.
* Run `su -` to switch to the root user account.

> Use the following guidelines when you document commands that require root privileges:

Reviewer: ```suggestion
Use the following guidelines when your document commands that require root privileges:
```

If you include a shell prompt to indicate that a user with root privileges must run the command, also include a statement about this requirement in the step text, the introductory text, or the prerequisites.
====
+
* If a command requires a temporary switch to root privileges, use `sudo` at the beginning of the sample command syntax. 
If the sample command includes a shell prompt together with `sudo`, the prompt should remain `$` and not be changed to `#`, as in the following example: 
+
[source,terminal]
----
$ sudo systemctl start firewalld
----
+
* If multiple commands in a procedure require root privileges, add introductory content to tell the user about the requirement. 
The following example shows one way that a writer might integrate a requirement for root access into introductory text for a procedure:
+
.Example AsciiDoc
----

> A user must perform some of these tasks as the root user or by acquiring effective root privileges using a _setuid_ program, such as `sudo` or `su`.

Reviewer 2: > or by acquiring effective root privileges using a _setuid_ program, such as `sudo` or `su`.

The mention of `setuid` here seems a bit out of place. Since we're primarily dealing with the enterprise space, I think we should probably provide guidance to use `sudo` by default, and `su` only at the insistence of the SME reviewing the relevant content. 

For context: 
sudo allows a permitted user to execute a command as the superuser or another user, as specified by the security policy.  Security policies may support credential caching to allow the user to run sudo again for a period of time without requiring authentication.  By default, the sudoers policy caches credentials on a per-terminal basis for 5 minutes. 

su allows commands to be run with a substitute user and group ID. When called with no user specified, su defaults to running an interactive shell as root.

Author: I agree that mentioning `setuid` here is not needed, and I've updated the PR to remove it. (I actually took this example from the current RHEL docs, btw.) I don't think it's necessary to go into too much detail about the differences between `su` vs `sudo` here because we link to more in-depth info in the "Additional resources" section. This text is meant as a generic example and is not meant to be prescriptive; it's at the writer's discretion how they want to recommend gaining root privileges.

Reviewer 2: The "Additional resources" info linked says (unfortunately formatted, imo -- nestled under the embedded video but above visual links): 

> But when do you use one, not another? Since the sudo policy is defined in /etc/sudoers, this can give powerful permission controls. **Since sudo can pretty much do everything that su can, I would say it is best to stick with sudo unless you are working with some legacy codes that require the su command.**

>  it's at the writer's discretion how they want to recommend gaining root privileges.

I understand that's how its presented here currently. I don't think leaving it to writer discretion is the most customer-focused approach, particularly without at least a warning that some commands when executed as root (via `su`) can cause catastrophic failure & to take special care to have a SME sign off on its use over `sudo`.

If you include a shell prompt to indicate that a user with root privileges must run the command, also include a statement about this requirement in the step text, the introductory text, or the prerequisites.
====
+

> * If a sample command includes a shell prompt together with `sudo`, do not change the prompt to `#`. Keep the `$` prompt, as shown in the following example: 

Reviewer 3: ```suggestion
* Use the `$` prompt, not `#`, when a sample command includes `sudo`, as shown in the following example: 
```

Author: Great suggestion. Made the change with a slight revision.Thanks.

+
[source,terminal]
----
$ sudo systemctl start firewalld
----
+
* If multiple commands in a procedure require root privileges, add introductory content to tell the user about the requirement. 

> The following example shows one way that a writer might integrate a requirement for root access into introductory text for a procedure:

Reviewer 3: ```suggestion
Here is an example of how to integrate a root access requirement into the introduction for a procedure:
```

Author: Thanks. I try to avoid starting sentences with "here is", "there is", or "there are" because these phrases get through the subject and verb of the sentence without really saying very much. "root access requirement" is also too noun-heavy for my tastes.

But I've edited other parts of the sentence to shorten and make more "you"-centric.

+
.Example AsciiDoc
----

> You must perform some of these tasks by acquiring root privileges, which you can do, for example, by prepending the `sudo` command to the command syntax.

Author: Good point. Changed per your suggestion.

> You must perform some of these tasks by acquiring root privileges, which you can do, for example, by adding the `sudo` command to the beginning of the command syntax.

Reviewer 3: ```suggestion
Some tasks in this procedure require root privileges, which you can get by prefixing commands with `sudo`.
```

Author: Good suggestion! Edited with your changes in minde.
